test/e2e/volumes.go

```diff
@@ -170,6 +171,14 @@ func volumeTestCleanup(client *client.Client, config VolumeTestConfig) {
 	}
 
 	if config.serverImage != "" {
+		if err := waitForPodTerminatedInNamespace(client, config.prefix+"-client", "", config.namespace); !apierrs.IsNotFound(err) {
```

Comment:
@yujuhong yes, that is what I see in the tests with this fix: the client pods are deleted gracefully and volumes are unmounted.